### ISSUE TITLE
Sampling question for small dataset

### ISSUE THREAD

**Issue by lukemunn** [jcjohnson/torch-rnn#118]
Thanks for this great and very interesting library. This is not so much a code issue as a question (and perhaps newbie one at that).

Just wondering how the original sampling via the preprocessor affects the output?

I have quite a small dataset for NNs (subset of afghan war diaries), and it seems the default 10% sampling might be really restricting the variety of output I'm getting. Grammatically the sample sentences are quite good but repeat themselves a lot and basically replicate much of the copy without reconfiguring it in 'creative' ways. Tried upping this to 100% in the preprocessor but then got errors when trying to train. 

Thoughts or suggestions much appreciated.

**Comment by Neil-Aframe** [comment]
@lukemunn: The 10% value is not a sample for training, but a sample for testing (so that you can see progress reported as the loss value)

The docs are https://github.com/jcjohnson/torch-rnn/blob/master/doc/flags.md#preprocessing

and the flags are:

> - `--val_frac`: What fraction of the data to use as a validation set; default is `0.1`.
> - `--test_frac`: What fraction of the data to use as a test set; default is `0.1`.

so by default you train on 80% of the original data. You could reduce those values to train on more of the data, but it probably won't make a significant difference to the output.

You might be able to see more variation by using a higher value for `-temperature` when sampling (this may introduce many non-words and other problems though), and you could maybe try some different network architectures. 

It does seem that the RNN approach for generating sequences is very data hungry.

**Comment by lukemunn** [comment]
OK thanks Neil for that clarification. 

Yes I can definitely see the potential with torch-rnn but unfortunately never quite get to the point where significant number of words are spelled correctly or feels like the model is mastered, even though loss gets down to 0.1 or less. Definitely not up to the level of karpathys Shakespeare, etc examples.

Anyway I might try training with 256 instead of default 128, and perhaps 3 layers, and see how it goes.
